lib/grammers-mtproto/src/transport/obfuscated.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright 2020 - developers of the `grammers` project.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+use grammers_crypto::{obfuscated::ObfuscatedCipher, DequeBuffer};
+use log::debug;
+
+use super::{Error, Tagged, Transport, UnpackedOffset};
+
+/// An obfuscation protocol made by telegram to avoid ISP blocks.
+/// This is needed to connect to the Telegram servers using websockets or
+/// when conecting to MTProto proxies (not yet supported).
+///
+/// It is simply a wrapper around another transport, which encrypts the data
+/// using AES-256-CTR with a randomly generated key that is then sent at the
+/// beginning of the connection.
+///
+/// Obfuscated transport can only be used with "tagged" transports, which
+/// provide a way to get the obfuscated tag that is used in the encryption.
+/// See the linked documentation for more information.
+///
+/// [Transport Obfuscation](https://core.telegram.org/mtproto/mtproto-transports#transport-obfuscation)
+pub struct Obfuscated<T: Transport + Tagged> {
+    inner: T,
+    head: Option<[u8; 64]>,
+    cipher: ObfuscatedCipher,
+}
+
+const FORBIDDEN_FIRST_INTS: [[u8; 4]; 7] = [
+    [0x48, 0x45, 0x41, 0x44], // "HEAD"
+    [0x50, 0x4f, 0x53, 0x54], // "POST"
+    [0x47, 0x45, 0x54, 0x20], // "GET "
+    [0x4f, 0x50, 0x54, 0x49], // "OPTI"
```

Comment:
Does this read better? Not sure. Either is probably fine.

```suggestion
    [b'H', b'E', b'A', b'D'], // HTTP HEAD
    [b'P', b'O', b'S', b'T'], // HTTP POST
    [b'G', b'E', b'T', b' '], // HTTP GET
    [b'O', b'P', b'T', b'I'], // HTTP OPTIONS
```

Reply:
Yeah, this is better.